### PR TITLE
Update local connection monitor on local connection update

### PR DESCRIPTION
**Pull request by Bolodya1997**
## Description
Update local connection monitor on local connection update

## Motivation and Context
If NSE has been changed during the healing, client does not receive an update for the opened local connection. It can lead to some problems if client behavior depends on connection parameters.

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [x] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Bolodya1997** [comment]
This PR is based on #1146 and so comes after.

File: .mk/k8s.mk [hunk 1]
```diff
@@ -24,7 +24,7 @@ DEPLOY_ICMP_KERNEL = icmp-responder-nse nsc
 DEPLOY_ICMP_VPP = vppagent-icmp-responder-nse vppagent-nsc
 DEPLOY_ICMP = $(DEPLOY_ICMP_KERNEL) $(DEPLOY_ICMP_VPP)
 DEPLOY_VPN = secure-intranet-connectivity vppagent-firewall-nse vppagent-passthrough-nse vpn-gateway-nse vpn-gateway-nsc
-DEPLOYS = $(DEPLOY_INFRA) $(DEPLOY_ICMP) $(DEPLOY_VPN)
+DEPLOYS = $(DEPLOY_INFRA) $(DEPLOY_ICMP) $(DEPLOY_VPN) monitoring-nsc
```
Comment (nickolaev) [inline .mk/k8s.mk]: Please add `monitor-nsc` to `DEPLOY_MONITOR`.

File: .mk/k8s.mk [hunk 2]
```diff
@@ -201,111 +202,46 @@ k8s-nsmgr-save:  $(addsuffix -save,$(addprefix ${CONTAINER_BUILD_PREFIX}-,$(NSMG
...
-k8s-vppagent-passthrough-nse-load-images:
-.PHONY: k8s-skydive-build
-k8s-skydive-build:
+.PHONY: k8s-icmp-responder-nse-build
+k8s-icmp-responder-nse-build: k8s-test-nse-build
 
-.PHONY: k8s-skydive-save
-k8s-skydive-save: k8s-skydive-build
+.PHONY: k8s-icmp-responder-nse-save
+k8s-icmp-responder-nse-save: k8s-test-nse-save
```
Comment (nickolaev) [inline .mk/k8s.mk]: These changes do not belong to this patch. Please separate them and explain what is the purpose.
I am also inclined to discuss a more radical change to our Makefile system.

**Comment by Bolodya1997** [comment]
@nickolaev 
Are you talking about creating a new `/test/application` directory with common docker images and so that all new testing-only applications will be located there?
Or do you mean that `examples` directory is no more need in `networkservicemesh/networkservicemesh` as we have `networkservicemesh/examples` and so it should be completely transformed to `/test/application` directory?

File: .mk/k8s.mk (hunk 1 above)
Comment (Bolodya1997) [reply .mk/k8s.mk]: My mistake, `monitor-nsc` is testing application so I have completely removed it from deployments : 741cbb5

File: .mk/k8s.mk (hunk 2 above)
Comment (Bolodya1997) [reply .mk/k8s.mk]: Reverted it : 741cbb5

**Comment by Bolodya1997** [comment]
@nickolaev 
Here is #1160 to rename `examples` to `test/applications` and so I've moved new `monitoring-nsc` to this new directory : 5eed0f8

**Comment by haiodo** [comment]
Ok, so we waiting for https://github.com/networkservicemesh/networkservicemesh/pull/1160 to be merged, before merging this one.

File: test/integration/recover_update_local_connection_test.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	nodesConf, err := kubetest.SetupNodes(k8s, 2, defaultTimeout)
+	Expect(err).To(BeNil())
+	defer kubetest.FailLogger(k8s, nodesConf, t)
+
+	nse1 := kubetest.DeployICMP(k8s, nodesConf[0].Node, "icmp-responder-nse-1", defaultTimeout)
+
+	nsc := kubetest.DeployMonitoringNSC(k8s, nodesConf[0].Node, "monitoring-nsc", defaultTimeout)
+	k8s.WaitLogsContains(nsc, "monitoring-nsc", "Monitor started", defaultTimeout)
+
+	time.Sleep(5 * time.Second) // we need to be sure that NSC is updated only with the initial connection
```
Comment (lobkovilya) [inline test/integration/recover_update_local_connection_test.go]: Looks like at that point we already know that logs contain "Monitor started" which is written only in case of receiving initial connection. So why do we need that Sleep?

Comment (lobkovilya) [reply test/integration/recover_update_local_connection_test.go]: Oh seems like to ensure that there are no more received events

File: .circleci/config.yml [hunk 4]
```diff
@@ -635,6 +635,11 @@ workflows:
           container: "nsc"
           requires:
             - sanity-check
+      - build-container:
+          name: "build-monitoring-nsc"
+          container: "monitoring-nsc"
+          requires:
+            - sanity-check
```
Comment (nickolaev) [inline .circleci/config.yml]: You probably want to update `docker-push-latest` too.

Comment (Bolodya1997) [reply .circleci/config.yml]: You are right, fixed this : c780836
